gtsfm/two_view_estimator.py

```diff
@@ -41,6 +53,46 @@ def __init__(self, matcher: MatcherBase, verifier: VerifierBase, corr_metric_dis
         self._verifier = verifier
         self._corr_metric_dist_threshold = corr_metric_dist_threshold
 
+    @classmethod
+    def bundle_adjust(
+        cls,
+        keypoints_i1: Keypoints,
+        keypoints_i2: Keypoints,
+        verified_corr_idxes: np.ndarray,
+        camera_intrinsics_i1: Cal3Bundler,
+        camera_intrinsics_i2: Cal3Bundler,
+        i2Ri1_initial: Optional[Rot3],
+        i2Ui1_initial: Optional[Unit3],
+    ) -> Tuple[Optional[Rot3], Optional[Unit3]]:
+
+        if i2Ri1_initial is None or i2Ui1_initial is None:
+            return None, None
+
+        # set the camera 2 pose as the global coordinate system
+        camera_i1 = PinholeCameraCal3Bundler(Pose3(i2Ri1_initial, i2Ui1_initial.point3()), camera_intrinsics_i1)
+        camera_i2 = PinholeCameraCal3Bundler(Pose3(), camera_intrinsics_i2)
+
+        # perform data association to construct 2-view BA input
+        ba_input, _ = DATA_ASSOC_2VIEW.run(
+            num_images=2,
+            cameras={0: camera_i1, 1: camera_i2},
+            corr_idxs_dict={(0, 1): verified_corr_idxes},
+            keypoints_list=[keypoints_i1, keypoints_i2],
+        )
+
+        ba_output = BUNDLE_ADJUST_2VIEW.run(ba_input)
+
+        # extract the camera poses
+        wPi1, wPi2 = ba_output.get_camera_poses()
+
+        if wPi1 is None or wPi2 is None:
+            logger.warning("2-view BA failed")
+            return i2Ri1_initial, i2Ui1_initial
+
+        i2Pi1_optimized = wPi2.between(wPi1)
```

Comment:
i prefer `i2Ti1` to `i2Pi1` since I think that's the convention we've generally been using